core/src/test/java/org/apache/calcite/sql/type/SqlTypeUtilTest.java

```diff
@@ -117,6 +120,34 @@
     SqlTypeCoercionRule.THREAD_PROVIDERS.set(defaultRules);
   }
 
+  @Test void testEqualAsCollectionSansNullability() {
+    final SqlTypeFactoryImpl factory =
+        new SqlTypeFactoryImpl(RelDataTypeSystem.DEFAULT);
```

Comment:
Use `f.typeFactory` instead.

Reply:
Thanks for your suggestion, updated already.